[BE-201] 레코드 쓴 날짜 반환

kdomo opened the pull request:
## 관련 이슈 번호
- [BE-201 / 레코드 쓴 날짜 반환](https://recodeit.atlassian.net/browse/BE-201)

## 설명
달력 버튼을 클릭 했을 때 로그인 된 사용자의 레코드 쓴 날짜를 반환해주는 API 입니다.
입력받은 년도와 월의 1일 부터 ~ 입력받은 년도와 월의 마지막일까지 between으로 처리하였습니다

## 변경사항

<!-- PR에 반영되어야 할 변경 사항들을 가능한 자세히 작성 해주세요. -->
<!-- - [x] 회원가입 및 로그인 -->

## 질문사항

kdomo committed:
[BE-201] test: 레코드 쓴 날짜 반환 API 테스트 코드 작성

kdomo committed:
[BE-201] feat: 레코드 쓴 날짜 반환 API

Jaeyeop-Jung commented on src/main/java/com/recordit/server/service/RecordService.java:
```diff
@@ -389,4 +396,32 @@ public RecordBySearchResponseDto getRecordsBySearch(
+		LocalDateTime start = getStartOfDay(standardDate.withDayOfMonth(1));
+		LocalDateTime end = getEndOfDay(standardDate.withDayOfMonth(standardDate.lengthOfMonth()));
+
+		TreeSet<Integer> writtenRecordDays = new TreeSet<>();
+		writtenRecordDays.addAll(
+				recordRepository.findAllByWriterAndCreatedAtBetween(member, start, end)
+						.stream().map(
+								record -> record.getCreatedAt().getDayOfMonth()
+						).collect(Collectors.toSet())
+		);
```
```suggestion
		TreeSet<Integer> writtenRecordDays = recordRepository.findAllByWriterAndCreatedAtBetween(member, start, end)
				.stream()
				.map(record -> record.getCreatedAt().getDayOfMonth())
				.collect(Collectors.toCollection(TreeSet::new));
```
요렇게 줄일 수 있을 거 같네용

Jaeyeop-Jung commented on src/main/java/com/recordit/server/service/RecordService.java:
```diff
@@ -389,4 +396,32 @@ public RecordBySearchResponseDto getRecordsBySearch(
+		Member member = memberRepository.findById(userIdBySession)
+				.orElseThrow(() -> new MemberNotFoundException("회원 정보를 찾을 수 없습니다."));
+
+		LocalDate standardDate = LocalDate.of(
+				writtenRecordDayRequestDto.getYearMonth().getYear(),
+				writtenRecordDayRequestDto.getYearMonth().getMonth(),
+				1
+		);
+
+		LocalDateTime start = getStartOfDay(standardDate.withDayOfMonth(1));
+		LocalDateTime end = getEndOfDay(standardDate.withDayOfMonth(standardDate.lengthOfMonth()));
```
이 부분을 DateTimeUtil에 메소드를 정의해서 빼는 방법으로 가면 좋을거 같네요

kdomo committed:
[BE-201] fix: date 로직 DateTimeUtil로 이동

kdomo committed:
Merge branch 'develop' into feature/BE-201

kdomo committed:
[BE-201] fix: 테스트를 위한 고정된 세션 삭제

kdomo committed:
[BE-201] fix: YearMonth -> LocalDateTime으로 통일
- WrittenRecordDayRequestDto.java 파일 필드 값 YearMonth에서 LocalDateTime으로 통일 하였습니다
- RecordController.java 파일 getWrittenRecordDays 메서드 반환 값 Set<Integer>에서 WrittenRecordDayResponseDto로 변경하였습니다

kdomo committed:
[BE-201] fix: LocalDateTime -> YearMonth로 변경
LocalDateTime 필드에 `@DateTimeFormat(pattern = "yyyy-MM")`을 활용하여 yyyy-MM 형식으로 받아보려고 하였으나
IllegalArgumentException 발생하여 YearMonth로 다시 변경하였습니다.

ErrorMessage: Failed to convert value of type 'java.lang.String' to required type 'java.time.LocalDateTime'
